common/thorhelper/thorxmlwrite.cpp

```diff
@@ -70,6 +70,30 @@ void CommonXmlWriter::closeTag()
     flush(false);
 }
 
+void CommonXmlWriter::cutFrom(IInterface *location, StringBuffer& databuf)
+{
+    if (flusher)
+        throwUnexpected();
+
+    CommonXmlPosition *position = dynamic_cast<CommonXmlPosition *>(location);
```

Comment:
Maybe.  I had purposely made it opaque, not wanting callers to make any assumptions about how they could make use of the location from outside.  Other implementations could save very different bits of information.  But open for discussion... for future changes.